extensions/amp-gfycat/0.1/amp-gfycat.js

```diff
@@ -48,34 +74,233 @@ class AmpGfycat extends AMP.BaseElement {
   }
 
   /** @override */
-  layoutCallback() {
-    const gfyid = user().assert(
+  buildCallback() {
+    this.videoid_ = this.getVideoId_();
+
+    // Enable autoplay by default
+    if (!this.element.hasAttribute('noautoplay')) {
+      this.element.setAttribute('autoplay', '');
+    }
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('amp-img');
+    dev().assert(this.videoid_);
```

Comment:
for these two lines, pleas do:

`const videoId = dev().assertString(this.videoid_);`